realm/realm-library/src/objectServer/java/io/realm/SyncConfiguration.java

```diff
@@ -1052,48 +1086,35 @@ public SyncConfiguration build() {
                 }
             }
 
-            // Check if the user has an identifier, if not, it cannot use /~/.
-            if (serverUrl.toString().contains("/~/") && user.getIdentity() == null) {
-                throw new IllegalStateException("The serverUrl contains a /~/, but the user does not have an identity." +
-                        " Most likely it hasn't been authenticated yet or has been created directly from an" +
-                        " access token. Use a path without /~/.");
-            }
-
             // Set the default Client Resync Mode based on the current type of Realm.
             // Eventually RECOVER_LOCAL_REALM should be the default for all types.
             if (clientResyncMode == null) {
-                clientResyncMode = ClientResyncMode.RECOVER_LOCAL_REALM;
+                clientResyncMode = ClientResyncMode.MANUAL;
```

Comment:
can we add a FIXME if the sync  team is willing to re-enable `RECOVER_LOCAL_REALM`?